### PR TITLE
Homework24: Ekaterina Tsiunel

### PR DESCRIPTION
**Pull request by ETsiunel**

### CODE REVIEW

File: hw_24/source/locators.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    password_input_login = "#password"
+    submit_button_login = '#submit'
+
+
+class LogoutLocators:
+    logout_button = '#logout'
+
+
+class AddContactLocators:
+    URL_add_contact = "https://thinking-tester-contact-list.herokuapp.com/contactList"
```
Comment (Stml89) [inline hw_24/source/locators.py]: move it to constants or url file/module

File: .github/workflows/pylint.yaml [hunk 2]
```diff
@@ -22,4 +22,4 @@ jobs:
           pip install -r requirements.txt
       - name: Analysing the code with pylint
         run: |
-          pylint --disable=C0200,W0718,W0621,R0903,C0116,W0613,R0801,C0103,E0611,import-error,pointless-string-statement,logging-fstring-interpolation,logging-not-lazy,consider-using-f-string $(git ls-files '*.py')
+          pylint --disable=C0200,W0718,W0621,R0903,C0116,W0613,R0801,C0103,E0611,import-error,pointless-string-statement,logging-fstring-interpolation,logging-not-lazy,consider-using-f-string,redefined-outer-name,missing-function-docstring,too-many-statements,too-many-locals,too-few-public-methods,missing-class-docstring $(git ls-files '*.py')
```
Comment (Stml89) [inline .github/workflows/pylint.yaml:25]: oh my gosh, why are you actually using pylint, if you have disabled so many checks?)))

File: hw_24/tests/test_contacts.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+PASSWORD = random_char(11)
+
+
+def test_signup(driver):
+    driver.get(AddUserLocators.URL_add_user)
+    driver.find_element(By.CSS_SELECTOR, AddUserLocators.signup_button).click()
+
+    first_name = driver.find_element(By.CSS_SELECTOR, AddUserLocators.firstname_input_signup)
+    first_name.clear()
+    first_name.send_keys("TestFirstName")
```
Comment (Stml89) [inline hw_24/tests/test_contacts.py]:
```suggestion
    first_name.send_keys("TestFirstName"), "TestFirstName" - should be in test data
```

File: hw_24/tests/test_contacts.py [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+def random_char(char_num):
+    return ''.join(random.choice(string.ascii_letters) for _ in range(char_num))
+
+
+EMAIL = random_char(7) + '@test.com'
+PASSWORD = random_char(11)
+
+
+def test_signup(driver):
```
Comment (Stml89) [inline hw_24/tests/test_contacts.py]: you gave this method the name `test_*`, but it doesn't test anything =(

File: hw_24/tests/test_contacts.py [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+    password = driver.find_element(By.CSS_SELECTOR, LoginLocators.password_input_login)
+    password.clear()
+    password.send_keys(PASSWORD)
+
+    driver.find_element(By.CSS_SELECTOR, LoginLocators.submit_button_login).click()
+
+
+def test_add_new_contact(driver):
+    test_login(driver)
```
Comment (Stml89) [inline hw_24/tests/test_contacts.py]: test as a precondition?

File: hw_24/tests/test_contacts.py [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+        except TimeoutException:
+            pytest.fail("Timed out waiting for deleting contact")
+
+    else:
+        pytest.fail("No contacts available to delete.")
+
+
+if __name__ == "__main__":
+    pytest.main()
```
Comment (Stml89) [inline hw_24/tests/test_contacts.py]: you dont really need it

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw_24/tests/test_contacts.py (hunk 4 above)
Comment (ETsiunel) [reply hw_24/tests/test_contacts.py]: fixed this one and `test_login`, named as `test_precondition_*`

File: hw_24/tests/test_contacts.py (hunk 5 above)
Comment (ETsiunel) [reply hw_24/tests/test_contacts.py]: sorry, my mistake.
Fixed

File: hw_24/tests/test_contacts.py (hunk 6 above)
Comment (ETsiunel) [reply hw_24/tests/test_contacts.py]: fixed

File: hw_24/source/locators.py (hunk 1 above)
Comment (ETsiunel) [reply hw_24/source/locators.py]: fixed, added urls module

File: hw_24/tests/test_contacts.py (hunk 3 above)
Comment (ETsiunel) [reply hw_24/tests/test_contacts.py]: done

File: .github/workflows/pylint.yaml (hunk 2 above)
Comment (ETsiunel) [reply .github/workflows/pylint.yaml:25]: really I wanted to turn it off a long time ago :)) 
disabled with your permission

**Comment by ETsiunel** [comment]
ready for review

File: .github/workflows/pylint.yaml (hunk 2 above)
Comment (Stml89) [reply .github/workflows/pylint.yaml:25]: ahahaha

File: hw_24/source/test_data.py
Comment (Stml89) [inline hw_24/source/test_data.py:1]: it is a test date, so put in `test_data` folder

File: hw_24/source/urls.py
Comment (Stml89) [inline hw_24/source/urls.py:1]: it is a test date, so put in `test_data` folder

File: hw_24/tests/test_contacts.py [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+from selenium import webdriver
+from selenium.common import TimeoutException
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from webdriver_manager.chrome import ChromeDriverManager
+from hw_24.source.urls import Urls
+from hw_24.source.locators import (AddUserLocators, LoginLocators,
```
Comment (Stml89) [inline hw_24/tests/test_contacts.py:15]: Good import usage

**Review by Stml89 — APPROVED** [review]
Fix comments and you can merge

File: hw_24/source/urls.py
Comment (ETsiunel) [reply hw_24/source/urls.py:1]: done

File: hw_24/source/test_data.py
Comment (ETsiunel) [reply hw_24/source/test_data.py:1]: done

**Pull request merged**